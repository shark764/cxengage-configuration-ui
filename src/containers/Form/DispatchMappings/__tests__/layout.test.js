/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import DispatchMappingsForm from '../layout';

describe('<DispatchMappingsForm />', () => {
  it('renders create form', () => {
    const rendered = shallow(
      <DispatchMappingsForm
        name="mockName"
        description="mockDescription"
        type="mockType"
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update form', () => {
    const rendered = shallow(
      <DispatchMappingsForm
        name="mockName"
        description="mockDescription"
        type="mockType"
        id="0000-0000-0000-0000-0000"
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
