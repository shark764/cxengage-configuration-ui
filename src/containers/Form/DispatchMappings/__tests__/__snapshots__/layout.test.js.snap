// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DispatchMappingsForm /> renders create form 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    data-automation="nameInput"
    disabled={false}
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    data-automation="descriptionInput"
    disabled={false}
    inputType="text"
    label="Description"
    name="description"
  />
  <SelectField
    data-automation="interactionList"
    disabled={false}
    label="Interaction Type *"
    name="channelType"
    options={
      Array [
        Object {
          "label": "Any",
          "value": "any",
        },
        Object {
          "label": "Voice",
          "value": "voice",
        },
        Object {
          "label": "Sms",
          "value": "sms",
        },
        Object {
          "label": "Email",
          "value": "email",
        },
        Object {
          "label": "Messaging",
          "value": "messaging",
        },
        Object {
          "label": "Work Item",
          "value": "work-item",
        },
      ]
    }
    required={true}
  />
  <SelectField
    data-automation="mappingList"
    disabled={false}
    label="Mapping *"
    name="interactionField"
    options={
      Array [
        Object {
          "label": "Customer",
          "value": "customer",
        },
        Object {
          "label": "Contact Point",
          "value": "contact-point",
        },
        Object {
          "label": "Integration",
          "value": "source",
        },
        Object {
          "label": "Direction",
          "value": "direction",
        },
      ]
    }
    required={true}
  />
  <SelectField
    data-automation="flowList"
    disabled={false}
    label="Dispatch to flow *"
    name="flowId"
    required={false}
  />
  <SelectField
    data-automation="versionList"
    disabled={false}
    label="Flow Version *"
    name="version"
    required={false}
  />
</form>
`;

exports[`<DispatchMappingsForm /> renders update form 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    data-automation="nameInput"
    disabled={false}
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    data-automation="descriptionInput"
    disabled={false}
    inputType="text"
    label="Description"
    name="description"
  />
  <SelectField
    data-automation="interactionList"
    disabled={false}
    label="Interaction Type *"
    name="channelType"
    options={
      Array [
        Object {
          "label": "Any",
          "value": "any",
        },
        Object {
          "label": "Voice",
          "value": "voice",
        },
        Object {
          "label": "Sms",
          "value": "sms",
        },
        Object {
          "label": "Email",
          "value": "email",
        },
        Object {
          "label": "Messaging",
          "value": "messaging",
        },
        Object {
          "label": "Work Item",
          "value": "work-item",
        },
      ]
    }
    required={true}
  />
  <SelectField
    data-automation="mappingList"
    disabled={false}
    label="Mapping *"
    name="interactionField"
    options={
      Array [
        Object {
          "label": "Customer",
          "value": "customer",
        },
        Object {
          "label": "Contact Point",
          "value": "contact-point",
        },
        Object {
          "label": "Integration",
          "value": "source",
        },
        Object {
          "label": "Direction",
          "value": "direction",
        },
      ]
    }
    required={true}
  />
  <SelectField
    data-automation="flowList"
    disabled={false}
    label="Dispatch to flow *"
    name="flowId"
    required={false}
  />
  <SelectField
    data-automation="versionList"
    disabled={false}
    label="Flow Version *"
    name="version"
    required={false}
  />
</form>
`;
